### PR TITLE
Added the optional stats_mode configuration for printing heap stats on a timer

### PR DESCRIPTION
**Pull request by MokoSan**
Added the optional ``stats_mode`` configuration for printing heap stats on a timer.  

The format of the period of the timer is given by a period magnitude that's an ``int`` and a period type that's either ``m`` to represent minutes or ``s`` to represent seconds. 

An example of the configuration is: 

```
stats_mode:
    # Print the heap stats on a timer with m: minutes; s: seconds.
    timer: 30s
```

If the stats_mode isn't specified, the user can still press `s` to manually display the heap stats.

### CODE REVIEW

**Comment by MokoSan** [comment]
As a heads up, this PR extends the behavior @NickCraver added for https://github.com/Maoni0/realmon/pull/3.

File: src/Program.cs [hunk 1]
```diff
@@ -10,6 +10,7 @@
 using System.Threading.Tasks;
 using realmon.Configuration;
 using realmon.Utilities;
+using System.Linq;
```
Comment (Maoni0) [inline src/Program.cs]: if you included Linq just to use string.Last() I would definitely not include it and use some other convenient way instead :) eg, `timerAsString[^1]`. yes we only do this once but to include a whold new namespace just to do this seems not really worth it (plus I don't want to encourage linq usage in this project in general :D)

File: src/Program.cs [hunk 2]
```diff
@@ -103,6 +105,43 @@ public static void RealTimeProcessingByProcessId(int pid, Options options, Confi
...
+            TimerCallback timerCallback = (_) =>
+            {
+                if (lastGC != null)
+                {
+                    PrintLastStats();
+                }
+            };
+
+            // TODO: Figure out when to dispose this.
+            heapStatsTimer = new Timer(callback: timerCallback,
```
Comment (Maoni0) [inline src/Program.cs:139]: I don't think you need to dispose it - if you are using the period stats mode you'd need this timer as long as the process is running anyway.

**Comment by MokoSan** [comment]
Thanks for the feedback! LOL - I used the range syntax in one spot but didn't in the other; glad you caught that!

**Review by Maoni0 — APPROVED** [review]
LGTM!

**Comment by Maoni0** [comment]
thanks @MokoSan! I'll merge this now. I'm actually taking the rest of the day off so I will look at your other PR tomorrow. so glad to see your contributions! 🙂

**Pull request merged**